### ISSUE TITLE
Install rust failed

### ISSUE THREAD

**Issue by nasyxx** [Homebrew/homebrew-core#22776]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

# Please always follow these steps:
- [X] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X] Ran `brew update` and retried your prior step?
- [X] Ran `brew doctor`, fixed all issues and retried your prior step?
- [X] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [X] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
- What happened (include command output)
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew install` commands)

Sorry to bother you. Here is the gist-logs https://gist.github.com/nasyxx/5d3913944ecd0454cf2193577a383c9a

I was trying to install rust with Racer code completion tool. `brew install rust --with-racer`. However, it failed.

I am not sure, is this a hard link problem?

# Formula additions or changes
To get formulae added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
We will close issues requesting formulae changes.

**Comment by ilovezfs** [comment]
Is it repeatable if you try again?

**Comment by nasyxx** [comment]
It still failed.

```
   ...
   Compiling cssparser-macros v0.3.0
   Compiling html5ever v0.18.0
   Compiling kuchiki v0.5.1
   Compiling html-diff v0.0.4
   Compiling rustdoc-tool v0.0.0 (file:///private/tmp/rust-20180112-54386-1jobawo/rustc-1.23.0-src/src/tools/rustdoc)
error: failed to link or copy `/private/tmp/rust-20180112-54386-1jobawo/rustc-1.23.0-src/build/x86_64-apple-darwin/stage2-tools/x86_64-apple-darwin/release/deps/
rustdoc_tool_binary-8c9dd52f4c133e30.dSYM` to `/private/tmp/rust-20180112-54386-1jobawo/rustc-1.23.0-src/build/x86_64-apple-darwin/stage2-tools/x86_64-apple-darw
in/release/rustdoc-tool-binary.dSYM`

Caused by:
  the source path is not an existing regular file


command did not execute successfully: "/private/tmp/rust-20180112-54386-1jobawo/rustc-1.23.0-src/build/x86_64-apple-darwin/stage0/bin/cargo" "build" "--target" "
x86_64-apple-darwin" "--release" "--manifest-path" "/private/tmp/rust-20180112-54386-1jobawo/rustc-1.23.0-src/src/tools/rustdoc/Cargo.toml"
expected success, got: exit code: 101


failed to run: /private/tmp/rust-20180112-54386-1jobawo/rustc-1.23.0-src/build/bootstrap/debug/bootstrap build
Build completed unsuccessfully in 1:31:22
make: *** [all] Error 1
```

**Comment by ilovezfs** [comment]
Is there anything unusual about your `/tmp`? (e.g. is it on another drive)

**Comment by nasyxx** [comment]
Though it is not on another drive, I am not sure. Is there any way I can expel this problem?

**Comment by nasyxx** [comment]
```
in /tmp
λ lsa
total 100K
drwxrwxrwt 24 root wheel  768 Jan 12 21:02 .
drwxr-xr-x  6 root wheel  192 Sep 26 17:44 ..
-rw-r--r--  1 root wheel   24 Jan 11 14:42 .CoreDisplayIOFBOpenInfo
drwx------  2 root wheel   64 Jan 11 14:42 .vnc-0
-rw-r--r--  1 root wheel    0 Jan 11 14:41 AlTest1.err
-rw-r--r--  1 root wheel    0 Jan 11 14:41 AlTest1.out
drwx------  2 Nasy wheel   64 Jan 11 23:10 CrashUpload-QkO0X
drwx------  2 Nasy wheel   64 Jan 11 21:36 CrashUpload-WuNOb
-rw-r--r--  1 root wheel 9.8K Jan 12 14:54 SurgeHelper.log
drwxr-xr-x  3 Nasy wheel   96 Jan 11 16:08 Training
-rw-rw-rw-  1 root wheel  68K Jan 12 15:42 adobegc.log
drwx------  3 Nasy wheel   96 Jan 11 14:42 com.apple.launchd.LMmNdsRYTr
drwx------  3 Nasy wheel   96 Jan 11 14:42 com.apple.launchd.bLM6MMRLZS
drwxr-xr-x  3 Nasy wheel   96 Jan 11 16:08 configure
drwxrwxrwx  4 Nasy wheel  128 Jan 11 23:10 dumps
-rw-r--r--  1 Nasy wheel 1.4K Jan 12 01:02 gameoverlayui.log
-rw-r--r--  1 Nasy wheel 2.1K Jan 12 01:02 gameoverlayui.log.last
-rw-r--r--  1 Nasy wheel   99 Jan 11 18:16 gensee.log
srwx------  1 Nasy wheel    0 Jan 11 14:53 mongodb-27017.sock
drwxr-xr-x  2 Nasy wheel   64 Jan 12 05:21 npm-6357-d9cf61c8
-rw-------  1 Nasy wheel  442 Jan 12 18:55 open_xoEzwK3s.txt
drwxr-xr-x  2 root wheel   64 Jan 11 14:41 powerlog
srwxr-xr-x  1 Nasy wheel    0 Jan 11 21:28 steam_chrome_shmem_uid501_spid31731
drwx------  3 Nasy wheel   96 Jan 12 13:06 tmux-501
```

```
in /private/tmp
λ lsa
total 100K
drwxrwxrwt 24 root wheel  768 Jan 12 21:02 .
drwxr-xr-x  6 root wheel  192 Sep 26 17:44 ..
-rw-r--r--  1 root wheel   24 Jan 11 14:42 .CoreDisplayIOFBOpenInfo
drwx------  2 root wheel   64 Jan 11 14:42 .vnc-0
-rw-r--r--  1 root wheel    0 Jan 11 14:41 AlTest1.err
-rw-r--r--  1 root wheel    0 Jan 11 14:41 AlTest1.out
drwx------  2 Nasy wheel   64 Jan 11 23:10 CrashUpload-QkO0X
drwx------  2 Nasy wheel   64 Jan 11 21:36 CrashUpload-WuNOb
-rw-r--r--  1 root wheel 9.8K Jan 12 14:54 SurgeHelper.log
drwxr-xr-x  3 Nasy wheel   96 Jan 11 16:08 Training
-rw-rw-rw-  1 root wheel  68K Jan 12 15:42 adobegc.log
drwx------  3 Nasy wheel   96 Jan 11 14:42 com.apple.launchd.LMmNdsRYTr
drwx------  3 Nasy wheel   96 Jan 11 14:42 com.apple.launchd.bLM6MMRLZS
drwxr-xr-x  3 Nasy wheel   96 Jan 11 16:08 configure
drwxrwxrwx  4 Nasy wheel  128 Jan 11 23:10 dumps
-rw-r--r--  1 Nasy wheel 1.4K Jan 12 01:02 gameoverlayui.log
-rw-r--r--  1 Nasy wheel 2.1K Jan 12 01:02 gameoverlayui.log.last
-rw-r--r--  1 Nasy wheel   99 Jan 11 18:16 gensee.log
srwx------  1 Nasy wheel    0 Jan 11 14:53 mongodb-27017.sock
drwxr-xr-x  2 Nasy wheel   64 Jan 12 05:21 npm-6357-d9cf61c8
-rw-------  1 Nasy wheel  442 Jan 12 18:55 open_xoEzwK3s.txt
drwxr-xr-x  2 root wheel   64 Jan 11 14:41 powerlog
srwxr-xr-x  1 Nasy wheel    0 Jan 11 21:28 steam_chrome_shmem_uid501_spid31731
drwx------  3 Nasy wheel   96 Jan 12 13:06 tmux-501
```

**Comment by jb2cool** [comment]
I'm having issues with rust too when installing via LinuxBrew on Ubuntu 16.04.3 and CentOS 7.4

I'm seeing;

curl: (22) The requested URL returned error: 404 Not Found
Error: Failed to download resource "rust--cargobootstrap"
Download failed: https://static.rust-lang.org/dist/2017-11-12/cargo-0.23.0-x86_64-apple-darwin.tar.gz

**Comment by ilovezfs** [comment]
@jb2cool this is Homebrew not Linuxbrew. The relevant PR there is https://github.com/Linuxbrew/homebrew-core/pull/5410

**Comment by jb2cool** [comment]
Ah sorry.

**Comment by ilovezfs** [comment]
@jb2cool no worries.

@nasyxx "good" news. I can reproduce the issue locally.

**Comment by ilovezfs** [comment]
OK this is https://github.com/rust-lang/cargo/issues/4671

**Comment by ilovezfs** [comment]
@nasyxx this is now fixed if you `brew update`

**Comment by nasyxx** [comment]
Thank you very much for helping to solve this problem

**Comment by iMichka** [comment]
Fixed in Linuxbrew (#5410), please update.

**Comment by ilovezfs** [comment]
@nasyxx you're welcome!
